### PR TITLE
Add CONTRIBUTING.md under SIG Auth

### LINKED ISSUE #5760: SIG Auth needs a CONTRIBUTING.md
Follow up item from sig auth annual report.

### PR DESCRIPTION
**Pull request by pravarag**
**Which issue(s) this PR fixes**:

Fixes #5760 

As part of this PR, we are adding a new CONTRIBUTING.md under sig-auth

### CODE REVIEW

**Comment by pravarag** [comment]
@liggitt @mikedanese kindly review this PR and let me know for any changes if required 🙂

File: sig-auth/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+- A [Contributor Guide][contrib-guide] - some
+  [kubernetes/kubernetes]-specific content, but lots of info for the
+  entire project
+- A [Contributor Cheat Sheet][contrib-cheatsheet] - lots of resources
+  and handy links
+
+SIG Auth has multiple areas you can contribute to. Those contributions
+can be in the form of code, documentation, support being involved in
+mailing list discussions, attending meetings, and more. This guide
+describes different major functional areas SIG Apps is involved in,
```
Comment (liggitt) [inline sig-auth/CONTRIBUTING.md:16]:
```suggestion
describes different major functional areas SIG Auth is involved in,
```

File: sig-auth/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  what a KEP should look like,
+  [kuberenetes/enhancements/keps/sig-auth] has some examples. If
+  you're not sure what "large" means, [come ask us](#contact)
+- We find it polite to apply the `do-not-merge/hold` label via the `/hold`
+  command when in doubt about whether/when a PR should merge. If we don't
+  explain why we're adding a `/hold`, this is usually because the PR will
+  cause changes to be deployed on merge, and we want the person responsible
+  for deploying and monitoring the changes to decide when to `/hold cancel`.
+  In some cases this is the PR author, in other cases this may be whomever
+  is on-call for prow.k8.io.  If you are unsure which, please ask.
```
Comment (liggitt) [inline sig-auth/CONTRIBUTING.md:43]: this doesn't really apply to sig-auth PRs... this sounds like it was copied from a test-infra contributing doc

File: sig-auth/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+get started:
+
+- [`label:"good first issue"`][good-first-issue] - triaged per [Good First Issue docs][good-first-issue-docs]
+- [`label:"help wanted"`][help-wanted] - triaged per [Help Wanted
+  docs][help-wanted-docs]
+  
+## Guides
+
+If you're not sure where to contribute or what any of these mean, please see
+[/README.md] for a brief description of the various codebases in this repo.
```
Comment (liggitt) [inline sig-auth/CONTRIBUTING.md:76]: is https://github.com/kubernetes/community/README.md a dead link?

File: sig-auth/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+- milestone: during which release cycle do we plan on working on this issue
+- `sig/foo`: which SIG owns this work
+- `area/foo`: which subproject or code is this issue related to
+- `kind/foo`: which kind of work is this issue describing
+- `priority/foo`: how important is this issue
+
+For example, an issue related to cleaning up and consolidating (`kind/cleanup`)
+release-related (`sig/release`) jobs and dashboards (`area/config`) for the
+v1.16 cycle (`milestone: v1.16`) that may not get completed by the end of the
+cycle if more important or more urgent arises (`priority/important-longterm`).
```
Comment (liggitt) [inline sig-auth/CONTRIBUTING.md:63]:
```suggestion
cycle if more important or more urgent work arises (`priority/important-longterm`).
```

File: sig-auth/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+We try to have a non-stale pool of issues that are available for new
+contributors who want to help out but aren't sure what to work on or where to
+get started:
+
+- [`label:"good first issue"`][good-first-issue] - triaged per [Good First Issue docs][good-first-issue-docs]
+- [`label:"help wanted"`][help-wanted] - triaged per [Help Wanted
+  docs][help-wanted-docs]
```
Comment (liggitt) [inline sig-auth/CONTRIBUTING.md:71]: are these links defined?

**Review by liggitt — COMMENTED** [review]
the content looks like a reasonable starting point, but it looks like a lot of links were copied from other links and not defined...

**Comment by pravarag** [comment]
> the content looks like a reasonable starting point, but it looks like a lot of links were copied from other links and not defined...

Thanks for your review @liggitt and yes I did take some references from other sigs' CONTRIBUTING.md . Let me correct the changes you have suggested and put it up for review again.

**Comment by pravarag** [comment]
@liggitt I've made some changes as per your suggestions. Kindly let me know if I can put in few more information related to `SIG Auth` using few other resources as I might have missed some valuable points here.

**Comment by liggitt** [comment]
looks like a reasonable starting point to me. @enj @ritazh do you have specific sig-auth bits you'd like to add?

File: sig-auth/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# Contributing to SIG Auth
+
+Welcome to contributing to SIG Auth.
+
+If you haven't seen them already, the Kubernetes project has:
+
+- A [Contributor Guide][contrib-guide] - some
```
Comment (ritazh) [inline sig-auth/CONTRIBUTING.md:7]: Missing links here? May borrow from https://github.com/kubernetes/community/blob/master/sig-cluster-lifecycle/CONTRIBUTING.md

File: sig-auth/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+provides an overview of the areas, and gives pointers on getting more
+involved in each area. Consider this a launching point or the start of
+a [choose your own
+adventure](https://en.wikipedia.org/wiki/Choose_Your_Own_Adventure)
+for SIG Auth.
+
+## Workflow
+
+Just like rest of the Kubernetes project, we also use the same PR and
+review based workflow. Which means [use of the CNCF CLA][cla], [code
```
Comment (ritazh) [inline sig-auth/CONTRIBUTING.md:26]: Missing link

File: sig-auth/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+review by reviewers and approvers listed in OWNERS files][owners], and
```
Comment (ritazh) [inline sig-auth/CONTRIBUTING.md:27]: Missing link

**Comment by ritazh** [comment]
@pravarag thank you for your contribution! 🎉 Good starting PR! Left few nits for missing links.

File: sig-auth/CONTRIBUTING.md (hunk 8 above)
Comment (pravarag) [reply sig-auth/CONTRIBUTING.md]: Added the missing link.

File: sig-auth/CONTRIBUTING.md (hunk 7 above)
Comment (pravarag) [reply sig-auth/CONTRIBUTING.md]: Updated.

File: sig-auth/CONTRIBUTING.md (hunk 6 above)
Comment (pravarag) [reply sig-auth/CONTRIBUTING.md]: Updated as per the suggestion.

**Comment by pravarag** [comment]
@ritazh @liggitt I've updated the CONTRIBUTING.md as per review comments. Kindly let me know for any other changes/updates required.

**Pull request merged**